app/queries/boards/get_own_last_boards_in_history.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Returns a list of boards that are the last in a board history and user's a creator
+# It means that the user has :destroy_board permission
+
+module Boards
+  class GetOwnLastBoardsInHistory
+    CREATOR_IDENTIFIER = :destroy_board
+
+    attr_reader :user_id
+
+    def initialize(user_id)
+      @user_id = user_id
+    end
+
+    def call
+      Board.where.not(id: Board.select(:previous_board_id).where.not(previous_board_id: nil))
+           .joins(board_permissions_users: :permission)
+           .where(permissions: { identifier: CREATOR_IDENTIFIER },
+                  board_permissions_users: { user_id: user_id })
+           .order(created_at: :desc)
```

Comment:
Perhaps, we should add the limit in order not to return too much boards

Reply:
I think we expect to get all boards (grouped by boards' history), maybe then It can be reworked using pages

Reply:
Yes, the pages would be the best :) I have told about the limit like a quick way, while we don't have such requirement from users